apps/site/assets/ts/stop/components/Departures.tsx

```diff
@@ -39,18 +39,38 @@ const filteredByModes = (
   );
 };
 
+// Only return the first 2 routes for small screens, or all of them if there are filters, showMore
+const filterForSmallScreen = (
+  routes: RouteWithDirections[],
+  modes: Mode[],
+  showMore: boolean
+): RouteWithDirections[] => {
+  if (modes.length > 0 || showMore) {
+    return routes;
+  } else {
+    return routes.slice(0, 2);
+  }
+};
+
 const Departures = ({
   routes,
   routesAndAlerts,
   stop,
   selectedModes,
   dispatch
 }: Props): ReactElement<HTMLElement> => {
+  const [showMore, setShowMore] = useState<boolean>(false);
   const isModeSelected = (mode: Mode): boolean => selectedModes.includes(mode);
 
   const handleModeClick = (mode: Mode): void => dispatch(clickModeAction(mode));
 
   const filteredRoutes = filteredByModes(allRoutes(routes), selectedModes);
+  // Only show the first 2 elements of the array (unless filtered or button clicked)
+  const shortFilteredRoutes = filterForSmallScreen(
+    allRoutes(routes),
+    selectedModes,
+    showMore
+  );
```

Comment:
issue: I'm noticing some weird interactions when I try to use the mode filters on mobile, I think this is causing it to show all routes even when I try to specify a mode. Maybe the logic needs to be something like `shortFilteredRoutes = selectedModes.length ? filteredRoutes : filterForSmallScreen(...)`? 

Might consider a deploy to dev-blue to get some feedback on the interaction.